tests/conftest.py

```diff
@@ -203,7 +205,13 @@ def application_process(
 
 @fixture(scope="session")
 def docker_client() -> DockerClient:
-    return docker.from_env()
+    tests_id = "".join(random.choice(string.ascii_uppercase) for _ in range(10))
+    docker_client = docker.from_env()
+    setattr(docker_client, "test_id", tests_id)
+    yield docker_client
+    docker_client.containers.prune(filters={"label": tests_id})
+    if len(docker_client.containers.list(filters={"label": tests_id})) > 0:
+        raise Exception("Container has not been properly pruned")
```

Comment:
Always give the problematic data when raising the exception, otherwise one cannot debug the problem based on it.
So store the problematic containers in a list, and include `repr(..)` of that list in the `Exception`.